script.js

```diff
@@ -1,4 +1,72 @@
+//win loss record
+var numOfWin = 0;
+var numOfLose = 0;
+var numOfDraw = 0;
+var winPercentage = 0;
+
+//User name
+var currentGameMode = "waiting for username";
+var userName = "";
+
+// Generate a random integer from 1 to max inclusive
+var generateRandomInteger = function (max) {
+  var randomDecimal = Math.random() * max;
+  var randomInteger = Math.floor(randomDecimal) + 1;
+  return randomInteger;
+};
+//Generate scissors, paper or stone
+var generateSPS = function () {
+  var numSPSInRepertoire = 3;
+  var random = generateRandomInteger(numSPSInRepertoire);
```

Comment:
Great that you declared a variable here to be more explicit. 
using `generateRandomInteger(3)` is also ok! :)